fml/trace_event.cc

```diff
@@ -22,6 +23,21 @@ namespace tracing {
 
 #if TIMELINE_ENABLED
 
+static AsciiTrie s_whitelist;
+
+// This is a macro to force inlining to help unoptimized code speed.
+#define FLUTTER_TIMELINE_EVENT(label, timestamp0, timestamp1_or_async_id, \
+                               type, argument_count, argument_names,      \
+                               argument_values)                           \
+  if (s_whitelist.Query(label)) {                                         \
+    Dart_TimelineEvent(label, timestamp0, timestamp1_or_async_id, type,   \
+                       argument_count, argument_names, argument_values);  \
+  }
+
+void TraceSetWhitelist(const std::vector<std::string>& whitelist) {
```

Comment:
There needs to be a no-op version of this for the other side of the `#ifdef` where `TIMELINE_ENABLED` is not defined.

Reply:
done